evil-collection.el

```diff
@@ -175,6 +193,55 @@ Evil Collection for that mode. More arguments may be added in the future, so
 functions added to this hook should include a \"&rest _rest\" for forward
 compatibility.")
 
+(defun evil-collection-define-key (state package map-sym &rest bindings)
+  "Wrapper for `evil-define-key*' with additional features.
+Filter keys on the basis of `evil-collection-key-whitelist' and
+`evil-collection-key-blacklist'. Store bindings in
+`evil-collection-bindings-record'."
+  (declare (indent defun))
+  (let* ((whitelist (mapcar 'kbd evil-collection-key-whitelist))
+         (blacklist (mapcar 'kbd evil-collection-key-blacklist))
+         (record (cdr-safe (assoc package evil-collection-bindings-record))))
+    (while bindings
+      (let ((key (pop bindings))
+            (def (pop bindings)))
+        (when (or (and whitelist (member key whitelist))
+                  (not (member key blacklist)))
+          (push (cons key def) record)
+          (eval-after-load package
+            `(condition-case err
+                 (evil-define-key* ',state ,map-sym ,key ',def)
+               (error
+                (message "evil-collection: Bad binding %s"
+                         '(evil-define-key* ',state ,map-sym ,key ',def))))))))
+    (setq record (nreverse record))
+    (if (assoc package evil-collection-bindings-record)
+        (setcdr (assoc package evil-collection-bindings-record) record)
+      (push (cons package record) evil-collection-bindings-record))
+    nil))
+
+(defun evil-collection-describe-all-bindings ()
+  "Print bindings made by evil-collection to separate buffer."
+  (interactive)
+  (let ((buf (get-buffer-create "*evil-collection*")))
+    (switch-to-buffer-other-window buf)
+    (with-current-buffer buf
+      (erase-buffer)
+      (org-mode)
+      (dolist (package evil-collection-bindings-record)
+        (insert "\n\n* " (symbol-name (car package)) )
+        (insert "
+| Key | Definition |
+|-----|------------|
+")
+        (dolist (binding (cdr package))
+          ;; Don't print nil definitions
+          (when (cdr binding)
+            ;; FIXME: Handle keys with | in them
+            (insert (format "| %s | %S |\n"
+                            (key-description (car binding)) (cdr binding)))))
+        (org-table-align)))))
```

Comment:
What about a tabulated-list instead of Org mode?

Reply:
I thought about that. I was thinking of this and related functions as being akin to automatically generated readmes, hence org mode. I think there's also better familiarity with org-mode in general, but I don't have a strong opinion here.